crates/neon/src/types_impl/bigint.rs

```diff
@@ -0,0 +1,377 @@
+//! Types for working with [`JsBigInt`].
+
+use std::{error, fmt, mem::MaybeUninit};
+
+use crate::prelude::NeonResult;
+use crate::{
+    context::Context,
+    handle::Handle,
+    result::ResultExt,
+    sys::{self},
+    types::JsBigInt,
+};
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+/// Indicates if a `JsBigInt` is positive or negative
+pub enum Sign {
+    Positive,
+    Negative,
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+/// Indicates a lossless conversion from a [`JsBigInt`] to a Rust integer
+/// could not be performed.
+///
+/// Failures include:
+/// * Negative sign on an unsigned int
+/// * Overflow of an int
+/// * Underflow of a signed int
+pub struct Error<T>(T);
+
+impl<T> Error<T> {
+    /// Get the lossy value read from a `BigInt`. It may be truncated,
+    /// sign extended or wrapped.
+    pub fn into_inner(self) -> T {
+        self.0
+    }
+}
+
+impl<T> fmt::Display for Error<T>
+where
+    T: fmt::Display,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Loss of precision reading BigInt ({})", self.0)
+    }
+}
+
+impl<T> error::Error for Error<T> where T: fmt::Display + fmt::Debug {}
+
+impl<T, E> ResultExt<T> for Result<T, Error<E>>
+where
+    E: fmt::Display,
+{
+    fn or_throw<'a, C: Context<'a>>(self, cx: &mut C) -> NeonResult<T> {
+        self.or_else(|err| cx.throw_range_error(err.to_string()))
+    }
+}
+
+impl JsBigInt {
+    pub const POSITIVE: Sign = Sign::Positive;
+    pub const NEGATIVE: Sign = Sign::Negative;
+
+    /// Creates a `BigInt` from an [`i64`].
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use neon::{prelude::*, types::JsBigInt};
+    /// # fn example(mut cx: FunctionContext) -> JsResult<JsBigInt> {
+    /// let value: Handle<JsBigInt> = JsBigInt::from_i64(&mut cx, 42);
+    /// # Ok(value)
+    /// # }
+    /// ```
+    pub fn from_i64<'cx, C>(cx: &mut C, n: i64) -> Handle<'cx, Self>
+    where
+        C: Context<'cx>,
+    {
+        let mut v = MaybeUninit::uninit();
+        let v = unsafe {
+            assert_eq!(
+                sys::create_bigint_int64(cx.env().to_raw(), n, v.as_mut_ptr(),),
+                sys::Status::Ok,
+            );
+
+            v.assume_init()
+        };
+
+        Handle::new_internal(Self(v))
+    }
+
+    /// Creates a `BigInt` from a [`u64`].
```

Comment:
I'd probably mildly prefer we have examples for every entry point, but if you think it's just noise for the reader of the docs I'm persuadable.